current/fyne/builtins_fyne.go

```diff
@@ -90,6 +90,40 @@ var Builtins_fyne = map[string]*env.Builtin{
 			return *env.NewNative(ps.Idx, win, "fyne-widget")
 		},
 	},
+	"progressbar": {
+		Argsn: 0,
+		Doc:   "Creates a Fyne progressbar widget",
+		Fn: func(ps *env.ProgramState, arg0 env.Object, arg1 env.Object, arg2 env.Object, arg3 env.Object, arg4 env.Object) env.Object {
+			win := widget.NewProgressBar()
+			return *env.NewNative(ps.Idx, win, "fyne-progressbar")
+		},
+	},
+	"fyne-progressbar//set-value": {
```

Comment:
method on widgets still need to be prefixed with `fyne-`, is this expected?

Reply:
You mean the "fyne" in "fyne-progressbar//set-value" probably? 

Yes that before // is a "kind" of value on which this function works (dispatches by kind of fist argument). When you return a native you still return "fyne-progressbar", so that it's distinguished from for example "gtp-progressbar". Kind of a native is also mobile outside of context so it's good that it carries the namespace with it.

But a Rye programmer never types this, it's just something that is returned. programmer uses progressbar function in fyne context, it returns native with kind "fyne-progressbar" and when programmer uses set-text on that object the right set-text gets called (there can be many). 

I hope I managed to explaing ... I am in a hurry and am not taht good at explainging this anyway :)